src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -142,6 +143,31 @@ public IModelMetadataProvider MetadataProvider
         }
 
         /// <summary>
+        /// Gets or sets the <see cref="IModelBinderFactory"/>.
+        /// </summary>
+        public IModelBinderFactory ModelBinderFactory
```

Comment:
Why is this property `public` and why does it have a setter? Seems to enable some new scenario but I'm missing it. Note this does _not_ support the same scenarios as changing the `IModelBinders` in `Controller.ModelBinders` would do w/ existing code -- at least not in a straightforward fashion.